portality/tasks/old_data_cleanup.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+from typing import Type
+
+from portality.background import BackgroundTask
+from portality.core import app
+from portality.dao import DomainObject
+from portality.decorators import write_required
+from portality.lib.es_queries import ES_DATETIME_FMT
+from portality.models import Notification, BackgroundJob
+from portality.tasks.helpers import background_helper
+from portality.tasks.redis_huey import schedule, long_running
+
+target_queue = long_running
+
+
+def _clean_old_data(domain_class: Type[DomainObject], retention_days,
+                    datetime_field='created_date',
+                    logger_fn=None, ):
+    if logger_fn is None:
+        logger_fn = print
+
+    last_retention_date = datetime.datetime.now() - datetime.timedelta(days=retention_days)
+    query = {
+        'query': {
+            'range': {
+                datetime_field: {
+                    'lte': last_retention_date.strftime(ES_DATETIME_FMT),
+                }
+            }
+        }
+    }
+
+    num_record = domain_class.hit_count(query)
+    logger_fn(f'remove [{domain_class.__name__}] -- {datetime_field} <= {last_retention_date}')
+    logger_fn(f'number of [{domain_class.__name__}][{num_record}] to be remove.')
+    domain_class.delete_by_query(query)
+
+
+def clean_all_old_data(bgjob_retention_days=180,
+                       noti_retention_days=180,
+                       logger_fn=None):
+    if logger_fn is None:
+        logger_fn = print
+
+    _clean_old_data(Notification, noti_retention_days, logger_fn=logger_fn)
+    _clean_old_data(BackgroundJob, bgjob_retention_days, logger_fn=logger_fn)
```

Comment:
I wonder if we can make this process more generic by specifying a list of models to be cleaned up, and the parameters for that looked up directly in config.

So, for example, in this file somewhere

```python
CLEANUP_ON = [Notification, BackgroundJob]
```

Then in this function

```python
for klazz in CLEANUP_ON:
    _clean_old_data(klazz, logger_fn=logger_fn)
```

Then in `settings.py` instead of `TASKS_BGJOB_RETENTION_DAYS` and `TASKS_NOTIFICATION_RETENTION_DAYS` we have something like

```python
TASK_DATA_RETENTION_DAYS = {
    "notification": 180,
    "background_job": 180
}
```

Which means in `_cleanup_old_data` you could just get the number of days with

```python
config.get("TASK_DATA_RETENTION_DAYS", {}).get(domain_class.__type__, 180)
```